src/compas/geometry/primitives/polyline.py

```diff
@@ -258,11 +259,134 @@ def transform(self, T):
+        """Splits a polyline at corners larger than the given angle_threshold
 
+        Parameters:
+        -----------
+        angle_threshold: radians
+
+        Returns
+        -------
+        list of polylines
+
+        """
+
+        corner_ids = []
+        split_polylines = []
+        points = self.points
+        seg_ids = list(range(len(self.lines)))
+
+        if self.is_closed():
+            seg_ids.append(0)
+
+        for seg1, seg2 in pairwise(seg_ids):
+            angle = self.lines[seg1].vector.angle(self.lines[seg2].vector)
+            if angle >= angle_threshold:
+                corner_ids.append(seg1+1)
+
+        if self.is_closed() and len(corner_ids) > 0:
+            if corner_ids[-1] != len(points):
+                corner_ids = [corner_ids[-1]] + corner_ids
+        else:
+            corner_ids = [0] + corner_ids + [len(points)]
+
+        for id1, id2 in pairwise(corner_ids):
+            if id1 < id2:
+                split_polylines.append(Polyline(points[id1:id2+1]))
+            else:
+                looped_pts = [points[i] for i in range(id1, len(points))] + points[1:id2+1]
+                split_polylines.append(Polyline(looped_pts))
+
+        return split_polylines
+
+    def tangent_at_point_on_polyline(self, point):
+        """Calculates the tangent vector of a point on a polyline
+
+        Parameters:
+        -----------
+        point: :class:`compas.geometry.Primitives.Point`
+
+        Returns
+        -------
+        Vector
+        """
+        for line in self.lines:
+            if is_point_on_line(point, line):
+                return line.direction
+
+    def divide_polyline(self, num_segments):
+        """Divide a polyline in equal segments
+
+        Parameters:
+        -----------
+        num_segments: int
+
+        Returns
+        -------
+        list
+            list of points :class:`compas.geometry.Primitives.Point`
+        """
+        segment_length = self.length/num_segments
+
+        num_pts = int(round(self.length/segment_length))
+        total_length = [0, 0]
+        division_pts = [self.points[0]]
+        new_polyline = self
+        for i in range(num_pts):
+            for i_ln, line in enumerate(new_polyline.lines):
+                total_length.append(total_length[-1] + line.length)
+                if total_length[-1] > segment_length:
+                    amp = (segment_length - total_length[-2]) / line.length
+                    new_pt = line.start + line.vector.scaled(amp)
+                    division_pts.append(new_pt)
+                    total_length = [0, 0]
+                    remaining_pts = new_polyline.points[i_ln+2:]
+                    new_polyline = Polyline([new_pt, line.end] + remaining_pts)
+                    break
+            if len(division_pts) == num_pts+1:
+                break
+
+        if len(division_pts) != num_pts+1:
+            division_pts.append(new_polyline.points[-1])
+
+        return division_pts
```

Comment:
This code is just the same as below, right?  so to avoid duplication, just call the function below.
```suggestion
        return self.divide_polyline_by_length(segement_length)
```

Reply:
Correct, I will add the edge case for the Polyline existing of only 2pts (a Line..) to `self.divide_polyline_by_length` and simplify `self.divide_polyline` as above.